templates/test/live.html.ep

```diff
@@ -1,24 +1,92 @@
+                % my $tab_count = $developer_session->ws_connection_count;
+                opened by <%= $developer_session->user->name %>
+                (started
+                    <abbr class="timeago" title="<%= $developer_session->t_created->datetime() %>Z">
+                        %= format_time($developer_session->t_created)
+                    </abbr>,
+                developer has <%= $tab_count %> browser <%= $tab_count == 1 ? 'tab' : 'tabs' %> open)
+            % }
+            % else {
+                no developer session opended
+                % if (is_admin) {
+                    - click to open
+                % }
+            % }
+        </div>
     </div>
-    <div id="running_module" class='col-sm-9'>Unknown</div>
-  </div>
-  % if (my $developer_session = $job->developer_session) {
-    <div class='row'>
-        <div class='col-sm-3'>
-        Developer session:
+    <div class="card-body">
+        <form action="#" method="get" id="developer-form" class="developer-mode-element" data-hidden-on="lockedByOtherDeveloper">
+            <p class="developer-mode-element" data-hidden-on="ownSession">
+                Select what you want to do and confirm using the button below. This will start a developer
+                session which will lock those developer options for other users.
+            </p>
+            <p class="developer-mode-element" data-visible-on="ownSession">
+                You started a developer session. Use the controls above to control the test behavior.
+            </p>
+            <h3>
+                Pause test execution
+                <%= help_popover('Help for <i>pause test execution</i>' => '
+                        <p>Pauses the test execution. So far it is only possible to pause at a
+                        specific module. The test can then be accessed via VNC for further
+                        investigation.</p>
+                        <p>Resuming the job after it has been paused is possible.</p>')
+                %>
+            </h3>
+            <div class="form-group row">
+                <label for="developer-pause-at-module" class="col-sm-3 col-form-label">
+                    at specific module
+                </label>
+                <div class="col-sm-9">
+                    %= include 'test/module_select'
+                </div>
+            </div>
+            <hr>
+            <div>
+                <a href="#" onclick="startDeveloperSession(); return false;"
+                    class="btn btn-warning developer-mode-element" data-hidden-on="ownSession">
+                    <i class="fas fa-lock"></i> Confirm & start developer session
+                </a>
+                <a href="#" onclick="quitDeveloperSession(); return false;"
+                    class="btn btn-danger developer-mode-element" data-visible-on="ownSession">
+                    <i class="far fa-stop-circle"></i> Cancel job
```

Comment:
Clicking on Cancel Job, will simply cancel the developer session. Better rename it to cancel developer session, and cancel the job. unless @coolo has a different opinion. But for me a test that was paused and likely touched, is tainted.